src/flat_bvh.rs

```diff
@@ -419,6 +421,131 @@ impl<T: BHValue + std::fmt::Display, const D: usize> BoundingHierarchy<T, D> for
+    ///
+    /// Note: the flat bvh is not optimized for nearest queries. It is recommended to use
+    /// a [`Bvh`] for nearest queries when possible.
+    /// The flat bvh will still be much faster than a linear search.
+    ///
+    /// [`FlatBvh`]: struct.FlatBvh.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bvh::aabb::{Aabb, Bounded};
+    /// use bvh::bounding_hierarchy::BoundingHierarchy;
+    /// use bvh::flat_bvh::FlatBvh;
+    /// use bvh::point_query::PointDistance;
+    /// use nalgebra::{Point3, Vector3};
+    /// use bvh::ray::Ray;
+    /// # use bvh::bounding_hierarchy::BHShape;
+    /// # pub struct UnitBox {
+    /// #     pub id: i32,
+    /// #     pub pos: Point3<f32>,
+    /// #     node_index: usize,
+    /// # }
+    /// #
+    /// # impl UnitBox {
+    /// #     pub fn new(id: i32, pos: Point3<f32>) -> UnitBox {
+    /// #         UnitBox {
+    /// #             id: id,
+    /// #             pos: pos,
+    /// #             node_index: 0,
+    /// #         }
+    /// #     }
+    /// # }
+    /// #
+    /// # impl Bounded<f32,3> for UnitBox {
+    /// #     fn aabb(&self) -> Aabb<f32,3> {
+    /// #         let min = self.pos + Vector3::new(-0.5, -0.5, -0.5);
+    /// #         let max = self.pos + Vector3::new(0.5, 0.5, 0.5);
+    /// #         Aabb::with_bounds(min, max)
+    /// #     }
+    /// # }
+    /// #
+    /// # impl BHShape<f32,3> for UnitBox {
+    /// #     fn set_bh_node_index(&mut self, index: usize) {
+    /// #         self.node_index = index;
+    /// #     }
+    /// #
+    /// #     fn bh_node_index(&self) -> usize {
+    /// #         self.node_index
+    /// #     }
+    /// # }
+    /// #
+    /// # impl PointDistance<f32,3> for UnitBox {
+    /// #     fn distance_squared(&self, query_point: Point3<f32>) -> f32 {
+    /// #         self.aabb().min_distance_squared(query_point)
+    /// #    }
+    /// # }
+    /// #
+    /// # fn create_bhshapes() -> Vec<UnitBox> {
+    /// #     let mut shapes = Vec::new();
+    /// #     for i in 0..1000 {
+    /// #         let position = Point3::new(i as f32, i as f32, i as f32);
+    /// #         shapes.push(UnitBox::new(i, position));
+    /// #     }
+    /// #     shapes
+    /// # }
+    ///
+    /// let point = Point3::new(0.0,0.0,0.0);
+    /// let mut shapes = create_bhshapes();
+    /// let flat_bvh = FlatBvh::build(&mut shapes);
+    /// let best_shape = flat_bvh.nearest_from(point, &shapes);
+    /// ```
+    ///
+    /// [`BoundingHierarchy`]: trait.BoundingHierarchy.html
+    /// [`Aabb`]: ../aabb/struct.Aabb.html
+    ///
+    fn nearest_from<'a, Shape: BHShape<T, D> + PointDistance<T, D>>(
+        &'a self,
+        query: Point<T, D>,
+        shapes: &'a [Shape],
+    ) -> Option<(&'a Shape, T)> {
+        if self.is_empty() {
+            return None;
+        }
+
+        let mut best_element = (&shapes[0], shapes[0].distance_squared(query));
```

Comment:
`shapes[0]` may suffer from the same issue as in the normal BVH (see previous comment)